src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/LastMinuteHandoverCaseService.kt

```diff
@@ -0,0 +1,153 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.response.LastMinuteHandoverCaseResponse
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.EligibilityService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.HdcService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.dates.ReleaseDateService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.CommunityManager
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.DeliusApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.fullName
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.IN_PROGRESS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.NOT_STARTED
+import java.time.Clock
+import java.time.LocalDate
+
+@Service
+class LastMinuteHandoverCaseService(
+  private val prisonerSearchApiClient: PrisonerSearchApiClient,
+  private val eligibilityService: EligibilityService,
+  private val licenceRepository: LicenceRepository,
+  private val hdcService: HdcService,
+  private val deliusApiClient: DeliusApiClient,
+  private val releaseDateService: ReleaseDateService,
+  @param:Value("\${hmpps.cases.last-minute.prisons}")
+  private val lastMinutePrisonCodes: List<String> = emptyList(),
+  private val overrideClock: Clock = Clock.systemDefaultZone(),
+) {
+
+  private data class LastMinuteReportDataBuilder(
+    val prisoners: Map<String, PrisonerSearchPrisoner>,
+    val inProgressEligiblePrisoners: Set<String> = emptySet(),
+    val eligiblePrisoners: Set<String> = emptySet(),
+    val licenceStartDates: Map<String, LocalDate> = emptyMap(),
+    val deliusData: Map<String, CommunityManager> = emptyMap(),
+    val candidates: Map<String, PrisonerSearchPrisoner> = emptyMap(),
+  )
+
+  fun getLastMinuteCases(): List<LastMinuteHandoverCaseResponse> {
+    log.info("Getting last minute handover cases")
+
+    val result = LastMinuteReportDataBuilder(getPrisonerData())
+      .withEligibleCandidates()
+      .withPreSubmissionState()
+      .withoutHdc()
+      .withDeliusData()
+      .withLicenceStartDates()
+
+    return result.toResponses()
+      .sortedWith(
+        compareBy<LastMinuteHandoverCaseResponse> { it.releaseDate }
+          .thenBy { it.prisonerName },
+      )
```

Comment:
Need to understand what order they need it in? might make more sense to be by region, prison code

Reply:
I think probably `prison => region => release date`